internal/replication/blockreplicator.go

```diff
@@ -780,14 +789,31 @@ Propose_Loop:
 			var isMembershipConfig bool
 
 			if httputils.IsConfigBlock(blockToPropose) {
-				// TODO verify the config-tx by itself and vs. current config
-				newClusterConfig := blockToPropose.Payload.(*types.Block_ConfigTxEnvelope).ConfigTxEnvelope.GetPayload().GetNewConfig()
+				configTxEnv := blockToPropose.GetConfigTxEnvelope()
+				// We validate a config TX preorder, to prevent a membership change to be applied to Raft, and then
+				// have the TX marked invalid during commit. Note that ConfigTxValidator.Validate reads the current
+				// config from the store. However,
+				//TODO together with the code to prevent in-flight config blocks before proposing a new one, the
+				// config from the store should match the config stored in the BlockReplicator.
+				// See: https://github.com/hyperledger-labs/orion-server/issues/281
+				valInfo, errVal := br.configTxValidator.Validate(configTxEnv)
+				if errVal != nil {
+					br.lg.Errorf("Failed to validate configTxEnv, internal error: %s", errVal)
+					br.releasePendingTXs(blockToPropose, "Declined to propose block, internal error in Validate", errVal)
+					continue Propose_Loop
+				}
+				if valInfo.Flag != types.Flag_VALID {
+					br.releasePendingTXs(blockToPropose, "Declined to propose block, invalid config tx",
+						&ierrors.BadRequestError{ErrMsg: fmt.Sprintf("Invalid config tx, reason: %s", valInfo.ReasonIfInvalid)})
```

Comment:
Isn't `continue Propose_Loop` should be here as well?

Reply:
Good catch!